packages/build/src/npm-packages.ts

```diff
@@ -0,0 +1,65 @@
+import { spawnSync } from 'child_process';
+import path from 'path';
+
+const PLACEHOLDER_VERSION = '0.0.0-dev.0';
+const PROJECT_ROOT = path.resolve(__dirname, '..', '..', '..');
+const LERNA_BIN = path.resolve(PROJECT_ROOT, 'node_modules', '.bin', 'lerna');
+
+export function bumpNpmPackages(version: string): void {
+  if (!version || version === PLACEHOLDER_VERSION) {
+    return;
+  }
+
+  spawnSync(LERNA_BIN, [
+    'version',
+    '--bump', version,
+    '--no-changelog',
+    '--no-push',
+    '--exact',
+    '--no-git-tag-version',
+    '--force-publish',
+    '--yes'
+  ], {
+    cwd: PROJECT_ROOT,
+    stdio: 'inherit'
+  });
+}
+
+export function publishNpmPackages(): void {
+  const packages = listNpmPackages();
+
+  const versions = Array.from(new Set(packages.map(({ version }) => version)));
+
+  if (versions.length !== 1) {
+    throw new Error(`Refusing to publish packages with multiple versions: ${versions}`);
+  }
+
+  if (versions[0] === PLACEHOLDER_VERSION) {
+    throw new Error('Refusing to publish packages with placeholder version');
+  }
+
+  spawnSync(LERNA_BIN, [
+    'publish',
+    '--bump', 'from-package',
+    '--no-changelog',
+    '--no-push',
+    '--exact',
+    '--no-git-tag-version',
+    '--force-publish',
+    '--yes'
+  ], {
+    cwd: PROJECT_ROOT,
+    stdio: 'inherit'
+  });
+}
+
+function listNpmPackages(): {version: string}[] {
+  const lernaListOutput = spawnSync(
+    LERNA_BIN, [
```

Comment:
Aren't you missing the `'list'` here? 🤔

Reply:
I really gotta find a way to test it out, at least manually once, i'm sure that the first time we use it, it won't work.